// flightplan/flightplan.js
var plan = require('./index');

module.exports = plan;

var HOST = 'brainbug.local';

plan.target('single', function(done) {
  setTimeout(function() {
    done([
      {
        host: HOST + 'lol',
        username: 'pstadler',
        agent: process.env.SSH_AUTH_SOCK,
        failsafe: true
      },
      {
        host: HOST,
        username: 'pstadler',
        agent: process.env.SSH_AUTH_SOCK,
      }
    ]);
  }, 1000);
}, { sudoUser: 'bar' });

plan.target('multi', [
  {
    host: HOST,
    username: 'pstadler',
    privateKey: '/Users/pstadler/.ssh/id_rsa',
    passphrase: ''
    //agent: process.env.SSH_AUTH_SOCK
  },
  {
    host: HOST,
    username: 'pstadler',
    privateKey: '/Users/pstadler/.ssh/id_rsa',
    passphrase: ''
    //agent: process.env.SSH_AUTH_SOCK
  }
]);

plan.remote(function(remote) {
  remote.ls();
});

plan.local(function(local) {
  console.log(plan.runtime.options);
  var foo = local.prompt('Hello?');
  console.log(foo);
  local.failsafe();
  local.ls('-al', {exec: { cwd: '/'}});
  local.transfer('lib/', '/tmp');
});

plan.remote(function(remote) {
  //remote.exec('read', {exec: {pty: true}});
  //remote.exec('tail -f /tmp/foo/index.js');
  // remote.silent();
  // remote.ls('-al');
  // remote.verbose();
  remote.ls('-al');
  // remote.exec('ls -al', { exec: { pty: true } });
  // remote.sudo('ls -al', { user: 'node' });
});

plan.local(function(local) {
  local.ls('-al foooo', {failsafe: true});
  //plan.abort();
});

plan.remote(function(remote) {
  remote.ls('-al');
});

// plan.local(function(local) {
//   local.with('cd /tmp', { failsafe: true }, function() {
//   	local.ls('-al bar');
//   	local.with('echo hello world', { failsafe: false }, function() {
//   	  local.ls('-al bar', { failsafe: true });
//   	});
//   });
//   var files = local.ls({ silent: true });
//   local.transfer(files, '/tmp/foo');
//   local.prompt('Do you really want to foo?', {hidden: true});
// });

// plan.local(function(local) {
//   local.log('this is a user message');
//   local.echo('hello world');
//   //plan.abort();
//   //local.prompt('foo', {hidden: true});
// });

// plan.local('default', function(local) {
//   local.exec('ls -al');
//   local.exec('ls foo', { failsafe: true });
//   try {
//   	local.exec('ls foo');
//   } catch(e) {
//   	console.log(e);
//   }
//   local.with('cd /tmp', { silent: true }, function() {
//   	local.ls();
//   })
// });

// plan.remote(function(remote) {
//   remote.silent();
//   remote.ls('-al');
//   remote.ls('foo', { failsafe: true })
//   remote.ls('foo');
//   remote.sudo('echo foo', { user: 'node' });
//   remote.verbose();
//   remote.with('cd /tmp', function() {
//   	remote.sudo('ls -al', { user: 'node' });
//   });
// });

// plan.remote(function(remote) {
//   remote.ls('-al /tmp/foo');
//   console.log(remote.target);
//   console.log(plan.runtime);
// });

//plan.run('default', 'multi', { debug: true });
